유방암의 여부가 True, False로 label

# 1. 필요한 모듈 import하기

In [1]:
from sklearn.datasets import load_breast_cancer
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report

# 2. 데이터 준비

### load_digits 메서드 사용

In [2]:
breast_cancer = load_breast_cancer()

In [3]:
breast_cancer.keys()
#print(dir(breast_cancer))

dict_keys(['data', 'target', 'frame', 'target_names', 'DESCR', 'feature_names', 'filename', 'data_module'])

# 3. 데이터 이해하기

### Feature Data 지정하기

In [4]:
# Feature Data 지정하기
breast_cancer_data = breast_cancer.data
print(breast_cancer_data.shape)

(569, 30)


### Label Data 지정하기

In [5]:
#Label Data 지정하기
breast_cancer_label = breast_cancer.target
breast_cancer_label

array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       0, 0, 1, 0, 1, 1, 1, 1, 1, 0, 0, 1, 0, 0, 1, 1, 1, 1, 0, 1, 0, 0,
       1, 1, 1, 1, 0, 1, 0, 0, 1, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1, 0, 0, 0,
       1, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 1, 1, 1, 1, 0, 1, 1, 0, 1,
       1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1, 0, 1, 0,
       0, 1, 0, 0, 1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1,
       1, 1, 0, 1, 1, 1, 1, 0, 0, 1, 0, 1, 1, 0, 0, 1, 1, 0, 0, 1, 1, 1,
       1, 0, 1, 1, 0, 0, 0, 1, 0, 1, 0, 1, 1, 1, 0, 1, 1, 0, 0, 1, 0, 0,
       0, 0, 1, 0, 0, 0, 1, 0, 1, 0, 1, 1, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0,
       1, 1, 1, 0, 1, 1, 1, 1, 1, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 0, 1, 1,
       1, 1, 0, 1, 1, 1, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 1, 1, 1, 1, 1, 1, 0, 1, 0, 1, 1, 0, 1, 1, 0, 1, 0, 0, 1, 1,
       1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0,

### Target Names 출력해 보기

In [6]:
# Target Names 출력해 보기
breast_cancer.target_names
# 0 이 malignant True
# 1 이 benign False

array(['malignant', 'benign'], dtype='<U9')

### 데이터 Describe 해보기

In [7]:
# 데이터 Describe 해 보기
print(breast_cancer.DESCR)

.. _breast_cancer_dataset:

Breast cancer wisconsin (diagnostic) dataset
--------------------------------------------

**Data Set Characteristics:**

    :Number of Instances: 569

    :Number of Attributes: 30 numeric, predictive attributes and the class

    :Attribute Information:
        - radius (mean of distances from center to points on the perimeter)
        - texture (standard deviation of gray-scale values)
        - perimeter
        - area
        - smoothness (local variation in radius lengths)
        - compactness (perimeter^2 / area - 1.0)
        - concavity (severity of concave portions of the contour)
        - concave points (number of concave portions of the contour)
        - symmetry
        - fractal dimension ("coastline approximation" - 1)

        The mean, standard error, and "worst" or largest (mean of the three
        worst/largest values) of these features were computed for each image,
        resulting in 30 features.  For instance, field 0 is Mean Radi

# 4. train, test 데이터 분리

In [8]:
import pandas as pd

### 문제지

In [9]:
#문제지
breast_cancer_pd = pd.DataFrame(data = breast_cancer_data, columns= breast_cancer.feature_names)
breast_cancer_pd

mean radius  mean texture  mean perimeter  mean area  mean smoothness  \
0          17.99         10.38          122.80     1001.0          0.11840   
1          20.57         17.77          132.90     1326.0          0.08474   
2          19.69         21.25          130.00     1203.0          0.10960   
3          11.42         20.38           77.58      386.1          0.14250   
4          20.29         14.34          135.10     1297.0          0.10030   
..           ...           ...             ...        ...              ...   
564        21.56         22.39          142.00     1479.0          0.11100   
565        20.13         28.25          131.20     1261.0          0.09780   
566        16.60         28.08          108.30      858.1          0.08455   
567        20.60         29.33          140.10     1265.0          0.11780   
568         7.76         24.54           47.92      181.0          0.05263   

     mean compactness  mean concavity  mean concave points  mean symmetry  \
0             0.27760         0.30010              0.14710         0.2419   
1             0.07864         0.08690              0.07017         0.1812   
2             0.15990         0.19740              0.12790         0.2069   
3             0.28390         0.24140              0.10520         0.2597   
4             0.13280         0.19800              0.10430         0.1809   
..                ...             ...                  ...            ...   
564           0.11590         0.24390              0.13890         0.1726   
565           0.10340         0.14400              0.09791         0.1752   
566           0.10230         0.09251              0.05302         0.1590   
567           0.27700         0.35140              0.15200         0.2397   
568           0.04362         0.00000              0.00000         0.1587   

     mean fractal dimension  ...  worst radius  worst texture  \
0                   0.07871  ...        25.380          17.33   
1                   0.05667  ...        24.990          23.41   
2                   0.05999  ...        23.570          25.53   
3                   0.09744  ...        14.910          26.50   
4                   0.05883  ...        22.540          16.67   
..                      ...  ...           ...            ...   
564                 0.05623  ...        25.450          26.40   
565                 0.05533  ...        23.690          38.25   
566                 0.05648  ...        18.980          34.12   
567                 0.07016  ...        25.740          39.42   
568                 0.05884  ...         9.456          30.37   

     worst perimeter  worst area  worst smoothness  worst compactness  \
0             184.60      2019.0           0.16220            0.66560   
1             158.80      1956.0           0.12380            0.18660   
2             152.50      1709.0           0.14440            0.42450   
3              98.87       567.7           0.20980            0.86630   
4             152.20      1575.0           0.13740            0.20500   
..               ...         ...               ...                ...   
564           166.10      2027.0           0.14100            0.21130   
565           155.00      1731.0           0.11660            0.19220   
566           126.70      1124.0           0.11390            0.30940   
567           184.60      1821.0           0.16500            0.86810   
568            59.16       268.6           0.08996            0.06444   

     worst concavity  worst concave points  worst symmetry  \
0             0.7119                0.2654          0.4601   
1             0.2416                0.1860          0.2750   
2             0.4504                0.2430          0.3613   
3             0.6869                0.2575          0.6638   
4             0.4000                0.1625          0.2364   
..               ...                   ...             ...   
564           0.4107                0.2216         

### 정답지

In [10]:
#정답지
breast_cancer_pd['label'] = breast_cancer.target
breast_cancer_pd

mean radius  mean texture  mean perimeter  mean area  mean smoothness  \
0          17.99         10.38          122.80     1001.0          0.11840   
1          20.57         17.77          132.90     1326.0          0.08474   
2          19.69         21.25          130.00     1203.0          0.10960   
3          11.42         20.38           77.58      386.1          0.14250   
4          20.29         14.34          135.10     1297.0          0.10030   
..           ...           ...             ...        ...              ...   
564        21.56         22.39          142.00     1479.0          0.11100   
565        20.13         28.25          131.20     1261.0          0.09780   
566        16.60         28.08          108.30      858.1          0.08455   
567        20.60         29.33          140.10     1265.0          0.11780   
568         7.76         24.54           47.92      181.0          0.05263   

     mean compactness  mean concavity  mean concave points  mean symmetry  \
0             0.27760         0.30010              0.14710         0.2419   
1             0.07864         0.08690              0.07017         0.1812   
2             0.15990         0.19740              0.12790         0.2069   
3             0.28390         0.24140              0.10520         0.2597   
4             0.13280         0.19800              0.10430         0.1809   
..                ...             ...                  ...            ...   
564           0.11590         0.24390              0.13890         0.1726   
565           0.10340         0.14400              0.09791         0.1752   
566           0.10230         0.09251              0.05302         0.1590   
567           0.27700         0.35140              0.15200         0.2397   
568           0.04362         0.00000              0.00000         0.1587   

     mean fractal dimension  ...  worst texture  worst perimeter  worst area  \
0                   0.07871  ...          17.33           184.60      2019.0   
1                   0.05667  ...          23.41           158.80      1956.0   
2                   0.05999  ...          25.53           152.50      1709.0   
3                   0.09744  ...          26.50            98.87       567.7   
4                   0.05883  ...          16.67           152.20      1575.0   
..                      ...  ...            ...              ...         ...   
564                 0.05623  ...          26.40           166.10      2027.0   
565                 0.05533  ...          38.25           155.00      1731.0   
566                 0.05648  ...          34.12           126.70      1124.0   
567                 0.07016  ...          39.42           184.60      1821.0   
568                 0.05884  ...          30.37            59.16       268.6   

     worst smoothness  worst compactness  worst concavity  \
0             0.16220            0.66560           0.7119   
1             0.12380            0.18660           0.2416   
2             0.14440            0.42450           0.4504   
3             0.20980            0.86630           0.6869   
4             0.13740            0.20500           0.4000   
..                ...                ...              ...   
564           0.14100            0.21130           0.4107   
565           0.11660            0.19220           0.3215   
566           0.11390            0.30940           0.3403   
567           0.16500            0.86810           0.9387   
568           0.08996            0.06444           0.0000   

     worst concave points  worst symmetry  worst fractal dimension  label  
0                  0.2654          0.4601                  0.11890      0  
1                  0.1860          0.2750                  0.08902      0  
2                  0.2430          0.3613                  0.08758      0  
3                  0.2575          0.6638                  0.17300      0  
4                  0.1625          0.2364                  0.07678   

### train, test 데이터 분리 

In [11]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(breast_cancer_data,
                                                   breast_cancer_label,
                                                   test_size = 0.3,
                                                   random_state = 10)



print('X_train 개수: ', len(X_train),', X_test 개수: ', len(X_test))

X_train 개수:  398 , X_test 개수:  171


In [12]:
X_train.shape, y_train.shape

((398, 30), (398,))

In [13]:
X_test.shape, y_test.shape

((171, 30), (171,))

In [14]:
y_train, y_test

(array([1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 0, 1, 1, 0, 0, 1, 1, 0, 1, 0, 1, 0,
        0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1, 1, 0,
        0, 1, 1, 1, 1, 1, 0, 1, 0, 1, 1, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 1,
        1, 1, 1, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 0, 0, 1, 1, 1,
        1, 1, 0, 1, 0, 1, 1, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0,
        1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0,
        1, 0, 1, 1, 0, 1, 1, 1, 1, 0, 1, 1, 0, 0, 1, 1, 0, 1, 0, 1, 1, 1,
        1, 1, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 1,
        1, 1, 0, 1, 0, 0, 1, 1, 1, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0,
        0, 1, 0, 1, 1, 0, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 1,
        1, 1, 0, 0, 1, 0, 1, 1, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1,
        1, 0, 1, 0, 1, 0, 0, 0, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 1, 0, 1, 1,
        0, 1, 1, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0,
        1, 1, 0, 0, 0, 1, 0, 1, 1, 1, 

# 5. 다양한 모델로 학습시켜보기

### Decision Tree

In [15]:
# Decision Tree
from sklearn.tree import DecisionTreeClassifier

decision_tree = DecisionTreeClassifier(random_state=32)
print(decision_tree._estimator_type)

decision_tree.fit(X_train, y_train)

y_pred = decision_tree.predict(X_test) # test 데이터로 예측
                                        # X_test데이터에는 정답인 label이 없고 
                                        # feature데이터만 존재
                                        # decision_tree 에 X_test 데이터로 predict
                                        # 실행하면 모델이 예측한 y_pred 도출
y_pred

classifier


array([0, 1, 1, 0, 1, 1, 1, 1, 0, 0, 1, 1, 0, 1, 0, 0, 1, 0, 1, 1, 0, 0,
       1, 1, 0, 0, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0,
       1, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 0, 1, 0, 1, 0, 0, 1, 0,
       1, 1, 0, 0, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 1, 1, 0, 0, 1, 1, 0, 0,
       1, 0, 1, 1, 0, 1, 1, 1, 0, 1, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 1, 1,
       1, 1, 0, 0, 0, 1, 1, 0, 0, 1, 0, 0, 1, 0, 1, 1, 0, 0, 1, 1, 1, 1,
       0, 1, 1, 0, 0, 0, 1, 0, 1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 0, 1, 0, 1,
       1, 1, 0, 1, 1, 1, 0, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1])

In [16]:
y_test

array([0, 1, 1, 0, 1, 1, 1, 1, 0, 0, 1, 1, 0, 1, 0, 0, 1, 1, 1, 1, 0, 0,
       1, 1, 0, 0, 1, 0, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0,
       1, 0, 0, 0, 0, 1, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 0, 1, 0, 0, 1, 1,
       1, 1, 0, 0, 1, 1, 1, 1, 1, 0, 1, 1, 0, 0, 1, 1, 0, 1, 1, 1, 1, 1,
       1, 1, 1, 1, 0, 1, 1, 1, 0, 1, 0, 1, 0, 0, 1, 1, 1, 1, 0, 0, 1, 1,
       1, 1, 0, 0, 0, 1, 0, 0, 0, 1, 1, 0, 1, 0, 1, 1, 0, 0, 1, 1, 1, 1,
       0, 1, 1, 0, 1, 0, 1, 0, 1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 0, 1, 0, 1,
       1, 1, 0, 1, 1, 0, 1, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1])

In [17]:
# 정확도
from sklearn.metrics import accuracy_score

accuracy = accuracy_score(y_test, y_pred)
accuracy

0.9122807017543859

### Random Forest 

In [18]:
# Random Forest 

from sklearn.ensemble import RandomForestClassifier

X_train, X_test, y_train, y_test = train_test_split(breast_cancer_data,
                                                   breast_cancer_label,
                                                   test_size = 0.3,
                                                   random_state = 10)


random_forest = RandomForestClassifier(random_state=32)
random_forest.fit(X_train, y_train)
y_pred = random_forest.predict(X_test)

print(classification_report(y_test, y_pred))

# 정확도
from sklearn.metrics import accuracy_score

accuracy = accuracy_score(y_test, y_pred)
accuracy

              precision    recall  f1-score   support

           0       0.95      1.00      0.98        59
           1       1.00      0.97      0.99       112

    accuracy                           0.98       171
   macro avg       0.98      0.99      0.98       171
weighted avg       0.98      0.98      0.98       171



0.9824561403508771

### SVM

In [19]:
# SVM
from sklearn import svm
svm_model = svm.SVC()

print(svm_model._estimator_type)

svm_model.fit(X_train, y_train)
y_pred = svm_model.predict(X_test)

print(classification_report(y_test, y_pred))

# 정확도
from sklearn.metrics import accuracy_score

accuracy = accuracy_score(y_test, y_pred)
accuracy

classifier
              precision    recall  f1-score   support

           0       0.94      0.86      0.90        59
           1       0.93      0.97      0.95       112

    accuracy                           0.94       171
   macro avg       0.94      0.92      0.93       171
weighted avg       0.94      0.94      0.93       171



0.935672514619883

### SGD Classifier

In [20]:
# SGD Classifier
from sklearn.linear_model import SGDClassifier
sgd_model = SGDClassifier()

print(sgd_model._estimator_type)


sgd_model.fit(X_train, y_train)
y_pred = sgd_model.predict(X_test)

print(classification_report(y_test, y_pred))

# 정확도
from sklearn.metrics import accuracy_score

accuracy = accuracy_score(y_test, y_pred)
accuracy

classifier
              precision    recall  f1-score   support

           0       0.80      0.93      0.86        59
           1       0.96      0.88      0.92       112

    accuracy                           0.89       171
   macro avg       0.88      0.90      0.89       171
weighted avg       0.90      0.89      0.90       171



0.8947368421052632

### Logistic Regression

In [21]:
# Logistic Regression
from sklearn.linear_model import LogisticRegression

logistic_model = LogisticRegression(max_iter=4000)


print(logistic_model._estimator_type)

logistic_model.fit(X_train, y_train)
y_pred = logistic_model.predict(X_test)

print(classification_report(y_test, y_pred))

# 정확도
from sklearn.metrics import accuracy_score

accuracy = accuracy_score(y_test, y_pred)
accuracy

classifier
              precision    recall  f1-score   support

           0       0.90      0.97      0.93        59
           1       0.98      0.95      0.96       112

    accuracy                           0.95       171
   macro avg       0.94      0.96      0.95       171
weighted avg       0.96      0.95      0.95       171



0.9532163742690059

# 6. 모델을 평가해보기

##### 모델을 평가해보기
##### 예측결과 어떻게 해석?
##### 성능을 평가하는 지표로는 무엇이 좋은지?    5가지
##### 평가지표 중 적절한것
##### 선택의 이유는?

Decision Tree 의 정확도 :        0.9122807017543859

Random Forest 의 정확도 :        0.9824561403508771

SVM 의 정확도 :                  0.935672514619883

SGD Classifier 의 정확도 :       0.8947368421052632

Logistic Regression  의 정확도 : 0.9532163742690059

## 예측결과==============


유방암의 여부를 진단하는것

정확도를 보았을때 랜덤숲 > 로지스틱 > SVM > 의사결정 나무 > SGD 순서

요소가 많으니 Random Forest를 사용하는게 좋을거같다

## =====================

In [22]:
# Random Forest 

from sklearn.ensemble import RandomForestClassifier

X_train, X_test, y_train, y_test = train_test_split(breast_cancer_data,
                                                   breast_cancer_label,
                                                   test_size = 0.3,
                                                   random_state = 10)


random_forest = RandomForestClassifier(random_state=32)
random_forest.fit(X_train, y_train)
y_pred = random_forest.predict(X_test)

print(classification_report(y_test, y_pred))

# 정확도
from sklearn.metrics import accuracy_score

accuracy = accuracy_score(y_test, y_pred)
accuracy

              precision    recall  f1-score   support

           0       0.95      1.00      0.98        59
           1       1.00      0.97      0.99       112

    accuracy                           0.98       171
   macro avg       0.98      0.99      0.98       171
weighted avg       0.98      0.98      0.98       171



0.9824561403508771

In [23]:
from sklearn.metrics import confusion_matrix

confusion_matrix(y_test, y_pred)

# TP 59
# FN 0
# FP 3
# TN 109

array([[ 59,   0],
       [  3, 109]], dtype=int64)

## 성능을 평가하는 지표로는 무엇이 좋은지?
## 평가지표 중 적절한것
## 선택의 이유는?

암 환자를 진단하는 모델에서는

Recall이 중요하다

양성을 음성으로 판단하면 안되기 때문

FN, 양성인데 음성으로 판단하는 경우가 적어야하니까

recall 은 분모에있는 FN이 낮을수록 커진다.

# 회고

#### 이번 프로젝트에서 어려웠던 점

#### 프로젝트를 진행하면서 알아낸 점 혹은 아직 모호한 점

#### 루브릭 평가 지표를 맞추기 위해서 시도한 것들

#### 루브릭 평가 관련 지표를 달성하지 못했을 때, 이유에 관한 추정. 느낌

#### 자기 다짐

### 1) 

이번 프로젝트를 진행하며 처음으로 접해보는 머신러닝과 사이킷런을 사용하는것에 어려움을 느꼈다. 

모든게 처음이다보니 모델의 개념이라던가 메소드의 사용법, 결과예측에 대해 보고 판단하고 학습하는것이 어려웠다

검색을 해보며 어려운 부분을 채워갔지만, 자주 검색해보고 학습하며 더 익숙해지도록 노력해야겠다.


### 2)

와인등급을 평가하는 2번째 프로젝트에서 SGD Classification 에서 

UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\SUN\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1334: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
C:\Users\SUN\miniconda3\lib\site-packages\sklearn\metrics\_classification.py:1334: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
  
  
이런 경고가 발생했었다.

이 경고는 레이블에 대해 계산할 f-score가 없음을 의미하고 이 경우 f-score는 0.0으로 간주되고, 평균 점수를 요청했으므로 계산에 0점을 포함했다는 점을 고려해야한다는 scikit-learn 경고문이라고 찾았다.

이때 import warnings
warnings.filterwarnings('ignore') 을 적어주어 경고문을 무시할 수 있었다

경고문의 이유와 표시되지 않는것은 찾았지만, 어떤부분에서 f-score가 0.0 이 나왔는지에 대해서는 완벽하게 파악하지 못해서 아쉬움이 있었다.

***
Logistic Regression에서 

ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

이러한 경고문이 발생했는데 이는 검색을 통해

 'max_iter = ' Gradient Descent 방식으로 최적의 해를 구하는 하이퍼 파라미터로 일정 수준이상이 되면 수렴되어 더이상 오르지 않는다.
 
 라는 답변을 찾을 수있었다. 
 해서 이를 LogisticRegression(max_iter=4000)로 max_iter 값을 4000 으로 주어 경고문을 피할 수 있었다.
 
max_iter값에 따라 결과를 도출해내는데에 시간이 적게 걸릴수도, 오래걸릴 수도 있다는 글을 보았지만 실험해보지않아 정확히 어떤 차이가 있는지 파악하지 못한것이 아쉬움에 남는다.
 
 
***
정확도는 직관적으로 모델 예측 성능을 나타내는 평가지표이지만, 이진 분류의 경우 데이터의 구성에 따라 ML모델의 성능을 왜곡할 수 있기때문에 정확도 수치 하나만으로 성능을 평가하지 않는다.

정확도를 평가 지표로 사용할 때는 불균형한 레이블값 분포에서 ML모델의 성능을 판단할 경우 정확도가 90%가 나오는 등 정확한 정확도를 얻을 수 없기때문에 

정확도만을 가지고 평가하기보다는 다른 분류 지표들과 함께 적용해야한다.

***
label이 많다는것이 정확히 몇개 정도가 되어야 많다고 하는것인지 그 표면상의 수치가 있지 않아서 의문이다.

### 3)

평가 지표를 맞추기위해 실습예제를 여러번 보고 최대한 비슷하게 코딩하며 실습해보았다.

경고문이 발생했을 경우, 경고문을 사라지게 하기위해 검색도 해보고 적용해보며 알지 못했던 개념을 익혔다.


### 4)

2번째 프로젝트에서 평가지표가 어떤게 좋을지 확실하게 판단하지 못해 애매하게 남겨둔것이 아쉬움에 남는다.

이는 아마 평가지표를 어느경우에 써야할지 완벽하게 학습하지 못한 이유이기 때문에 

평가지표를 선택하는것에 확실하게 평가를 하지 못한것 같다.

***
프로젝트가 시도해본것들이 많아 셀들이 깔끔하게 정리되어있지 않은것도 영향이 있을 것 같다.


### 5)

이번 프로젝트를 통해 새롭게 접해본것들이 많아 신기하면서도 흥미로웠지만, 한편으로는 기초부분일텐데 벌써 어려움을 겪는게 맞는건가 싶기도했다.

어려웠거나 헷갈렸던 부분들은 검색을 통해 답을 찾을 수 있어서 다행이였지만, 

아직 완벽하게 학습한게 아니라 더 공부하고 완전히 내껄로 만들고싶다.

첫번째 프로젝트는 잘 마무리 할 수 있었지만 다음 프로젝트는 잘 해낼 수 있을지, 어떤게 나올지 걱정 반 기대 반 인것 같다.

